homeassistant/components/aurora/__init__.py

```diff
@@ -126,5 +136,54 @@ async def _async_update_data(self):
 
         try:
             return await self.api.get_forecast_data(self.longitude, self.latitude)
-        except ConnectionError as error:
+        except ClientError as error:
             raise UpdateFailed(f"Error updating from NOAA: {error}") from error
+
+
+class AuroraEntity(CoordinatorEntity):
+    """Implementation of the base Aurora Entity."""
+
+    def __init__(
+        self,
+        coordinator: AuroraDataUpdateCoordinator,
+        name: str,
+        icon: str,
+    ):
+        """Initialize the Aurora Entity."""
+
+        super().__init__(coordinator=coordinator)
+
+        self._name = name
+        self._unique_id = f"{self.coordinator.latitude}_{self.coordinator.longitude}"
+        self._icon = icon
+
+    @property
+    def unique_id(self):
```

Comment:
My impression was this needs to be unique -- is this ok as long as there is only one entity per platform?

Reply:
It does need to be unique and it is unique based on latitude and longitude. While you can add multiple Aurora locations, there would be no reason to add multiple locations with the same lat/long combination.

Reply:
I think maybe i was worried that both the sensor and binary sensory have the same unique id.  I see the platform is part of the id: https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements